inference-engine/src/mkldnn_plugin/nodes/mkldnn_math_node.cpp

```diff
@@ -0,0 +1,271 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <cmath>
+#include <vector>
+#include <string>
+#include <mkldnn_types.h>
+#include "ie_parallel.hpp"
+#include <mkldnn_selective_build.h>
+#include "mkldnn_math_node.h"
+#include <nodes/common/tensor_desc_creator.h>
+#include <ngraph/opsets/opset2.hpp>
+#include "utils/general_utils.h"
+
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+bool MKLDNNMathNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (initializers.find(op->get_type_info()) == initializers.end()) {
+            errorMessage = "Unsupported Math layer type.";
+            return false;
+        }
+
+        if (MKLDNNPlugin::one_of(op->get_type_info(), ngraph::op::v0::HardSigmoid::type_info, ngraph::op::v0::Selu::type_info)) {
+            auto firstConst = ngraph::as_type_ptr<ngraph::op::v0::Constant>(op->get_input_node_shared_ptr(1));
+            auto secondConst = ngraph::as_type_ptr<ngraph::op::v0::Constant>(op->get_input_node_shared_ptr(2));
+            if (!firstConst || !secondConst) {
+                errorMessage = "Constant expected as the second and third inputs.";
+                return false;
+            }
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNMathNode::MKLDNNMathNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng,
+        MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache), alpha(0.f), beta(0.f), gamma(0.f) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    initializers[op->get_type_info()](op, *this);
+    typeOp = MKLDNNPlugin::one_of(op->get_type_info(), ngraph::op::v0::HardSigmoid::type_info, ngraph::op::v0::Selu::type_info) ? 3 : 1;
+}
+
+void MKLDNNMathNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    if (typeOp == 3) {
```

Comment:
You can generalize the logic by creating separate std::vector<DataConfigurator> with required number of  {TensorDescCreatorTypes::ncsp, Precision::FP32} elements. Number of inputs: getParentEdges().size()

Reply:
Fixed in 8c979d2088b5538bf6caf32c81e30a5b9dd485d1. Thank you!